### PR TITLE
ROX-17334: Move network graph checksum to blob store

### PR DESCRIPTION
**Pull request by c-du**
## Description
Move network graph checksum to blob store. There is no need for migration because the checksum will change during upgrade.

## Testing Performed

Existing test
Manual test for the normal operation. Tested with log and migration to verify update.

### CODE REVIEW

File: central/blob/datastore/datastore.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import (
 
 // Datastore provides access to the blob store
 type Datastore interface {
+	GetNames(ctx context.Context) ([]string, error)
```
Comment (connorgorman) [inline central/blob/datastore/datastore.go:13]: this is GetIDs now right?

File: central/networkgraph/entity/gatherer/load_bundled.go [hunk 2]
```diff
@@ -7,14 +7,42 @@ import (
...
+	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/networkgraph/defaultexternalsrcs"
 	"github.com/stackrox/rox/pkg/set"
 	"github.com/stackrox/rox/pkg/utils"
 )
 
-func loadBundledExternalSrcs(networkEntityDS entityDataStore.EntityDataStore) error {
+// loadLocalChecksum loads local checksum if it exists.
+func loadLocalChecksum(store blobstore.Datastore) (checksum []byte, err error) {
+	buf := bytes.NewBuffer([]byte{})
```
Comment (connorgorman) [inline central/networkgraph/entity/gatherer/load_bundled.go:22]: You can just say nil

File: central/networkgraph/entity/gatherer/load_bundled.go [hunk 3]
```diff
@@ -7,14 +7,42 @@ import (
...
 	entityDataStore "github.com/stackrox/rox/central/networkgraph/entity/datastore"
+	"github.com/stackrox/rox/generated/storage"
 	"github.com/stackrox/rox/pkg/networkgraph/defaultexternalsrcs"
 	"github.com/stackrox/rox/pkg/set"
 	"github.com/stackrox/rox/pkg/utils"
 )
 
-func loadBundledExternalSrcs(networkEntityDS entityDataStore.EntityDataStore) error {
+// loadLocalChecksum loads local checksum if it exists.
+func loadLocalChecksum(store blobstore.Datastore) (checksum []byte, err error) {
```
Comment (connorgorman) [inline central/networkgraph/entity/gatherer/load_bundled.go:21]: This is a small case so I don't think we should do named returns

File: central/networkgraph/entity/gatherer/online_gatherer.go [hunk 4]
```diff
@@ -29,23 +29,30 @@ var (
...
 	networkEntityDS entityDataStore.EntityDataStore
 	stopSig         concurrency.Signal
+	blobStore       blobstore.Datastore
 }
 
 // newDefaultExtNetworksGatherer returns an instance of NetworkGraphDefaultExtSrcsGatherer that reaches out internet to fetch the data.
 func newDefaultExtNetworksGatherer(networkEntityDS entityDataStore.EntityDataStore) NetworkGraphDefaultExtSrcsGatherer {
 	return &defaultExtSrcsGathererImpl{
 		networkEntityDS: networkEntityDS,
+		blobStore:       blobstore.Singleton(),
```
Comment (connorgorman) [inline central/networkgraph/entity/gatherer/online_gatherer.go:49]: I don't think this is the right place to put the singleton and should be wired into the gatherer singleton instead

**Review by connorgorman — COMMENTED** [review]
Merge the blob store one then request re-review please, but added some preliminary comments

File: central/networkgraph/entity/gatherer/online_gatherer.go [hunk 5]
```diff
@@ -143,3 +152,31 @@ func (g *defaultExtSrcsGathererImpl) reconcileDefaultExternalSrcs() error {
...
+	}
+	buf, _, exists, err := store.GetBlobWithDataInBuffer(blobAccessReadCtx, defaultexternalsrcs.LocalChecksumBlobPath)
+	if err != nil || !exists {
+		return nil, err
+	}
+	g.currentChecksum = buf.Bytes()
+	return g.currentChecksum, nil
+}
+
+func (g *defaultExtSrcsGathererImpl) writeChecksumLocally(store blobstore.Datastore, checksum []byte) error {
```
Comment (dashrews78) [inline central/networkgraph/entity/gatherer/online_gatherer.go:169]: I find this name confusing.  Looks to me that we are writing the checksum we found locally to the blobstore.  the name implies (at least to me) that we are writing it to the local file system.

**Review by connorgorman — APPROVED** [review]

**Pull request merged**